.github/workflows/build_all.yaml

```diff
@@ -130,177 +75,15 @@ jobs:
-
-  make_xilinx_examples:
-    needs: make_cava
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
-      with:
-        submodules: 'recursive'
-    - uses: cachix/install-nix-action@v12
-      with:
-        nix_path: nixpkgs=channel:nixos-unstable
-
-    - name: Create a "lockfile" from git submodule commits
-      id: get-submodule-ver
-      run: |
-        git submodule status > ./third_party/submodules.lock
-      shell: bash
-    - name: Cache third_party
-      id: third_party-cache
-      uses: actions/cache@v2
-      with:
-        path: third_party
-        key: ${{ runner.os }}-third_party-${{ hashFiles('third_party/submodules.lock') }}
-
-    - name: Use cava cache
-      uses: actions/cache@v2
-      with:
-        path: |
-          cava
-          /home/runner/.ghc/
-          /home/runner/.cabal/
-        key: ${{ runner.os }}-build-${{ github.run_id }}-cava
-    - run: nix-shell --run 'cd examples/xilinx; make -j1'
-
-  make_xilinx_tests:
-    needs: make_cava
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
-      with:
-        submodules: 'recursive'
-    - uses: cachix/install-nix-action@v12
-      with:
-        nix_path: nixpkgs=channel:nixos-unstable
-
-    - name: Create a "lockfile" from git submodule commits
-      id: get-submodule-ver
-      run: |
-        git submodule status > ./third_party/submodules.lock
-      shell: bash
-    - name: Cache third_party
-      id: third_party-cache
-      uses: actions/cache@v2
-      with:
-        path: third_party
-        key: ${{ runner.os }}-third_party-${{ hashFiles('third_party/submodules.lock') }}
-
-    - name: Use cava cache
-      uses: actions/cache@v2
-      with:
-        path: |
-          cava
-          /home/runner/.ghc/
-          /home/runner/.cabal/
-        key: ${{ runner.os }}-build-${{ github.run_id }}-cava
-    - run: nix-shell --run 'cd tests/xilinx; make -j1'
-  
-  make_firmware:
-    needs: make_cava
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
-      with:
-        submodules: 'recursive'
-    - uses: cachix/install-nix-action@v12
-      with:
-        nix_path: nixpkgs=channel:nixos-unstable
-
-    - name: Create a "lockfile" from git submodule commits
-      id: get-submodule-ver
-      run: |
-        git submodule status > ./third_party/submodules.lock
-      shell: bash
-    - name: Cache third_party
-      id: third_party-cache
+    - name: Cache cava
+      id: cava-cache
       uses: actions/cache@v2
       with:
-        path: third_party
-        key: ${{ runner.os }}-third_party-${{ hashFiles('third_party/submodules.lock') }}
+        path: cava
+        key: ${{ runner.os }}-cava-${{ hashFiles('third_party/submodules.lock', 'cava/**/*.v', 'cava/Makefile' ) }}
+    - name: Fail on cache miss
+      if: steps.third_party-cache.outputs.cache-hit != 'true' || steps.cava-cache.outputs.cache-hit != 'true'
+      run: exit 1
```

Comment:
Do I understand correctly that this `Fail on cache miss` step is just a safety check that should really always succeed, like an assert?

Reply:
Yes exactly, its probably unneeded but no harm in having it.